eng/pipelines/templates/steps/cache-maven-repository.yml

```diff
@@ -1,6 +1,17 @@
+parameters:
+  - name: JobType
+    type: string
+    default: test
+  - name: SDKType
+    type: string
+    default: client
+  - name: ServiceDirectory
+    type: string
+    default: 'not-specified' # Set a default that breaks in obvious ways.
+
 steps:
 - task: Cache@2
   inputs:
-    key: 'maven | $(CacheSalt) | $(Agent.OS) | $(Build.SourcesDirectory)/eng/versioning/external_dependencies.txt'
+    key: 'maven | "$(CacheSalt)" | "$(Agent.OS)" | $(Build.SourcesDirectory)/eng/versioning/external_dependencies.txt | "${{ parameters.SDKType }}" | "${{ parameters.ServiceDirectory }}" | "${{ parameters.JobType }}" | "$(TestFromSource)"'
```

Comment:
@alzimmermsft this seems a little more complex then it needs to be. Why not just add in `$(Agent.JobName)`? That should automatically capture all these other pivots because they are already part of the job name. The service directory isn't interesting because these cache's are already scoped by pipeline which is how we scope the service.